api/server/backend/build/backend.go

```diff
@@ -0,0 +1,75 @@
+package build
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/docker/distribution/reference"
+	"github.com/docker/docker/api/types"
+	backendtypes "github.com/docker/docker/api/types/backend"
+	"github.com/docker/docker/builder"
+	"github.com/docker/docker/builder/dockerfile"
+	"github.com/docker/docker/image"
+	"github.com/docker/docker/pkg/stringid"
+	"github.com/pkg/errors"
+	"golang.org/x/net/context"
+)
+
+// ImageComponent provides an interface for working with images
+type ImageComponent interface {
+	SquashImage(from string, to string) (string, error)
+	TagImageWithReference(image.ID, reference.Named) error
+}
+
+// Backend provides build functionality to the API router
+type Backend struct {
+	manager        *dockerfile.BuildManager
+	imageComponent ImageComponent
+}
+
+// NewBackend creates a new build backend from components
+func NewBackend(components ImageComponent, backend builder.Backend) *Backend {
```

Comment:
Are you suggesting to rename them, or that this should accept those things as parameters instead?

`backend` is not a builder, it's a `builderBackend`, so I can name it that.

The idea here is that components will grow as more of Builder is refactoring to split and shrink the backend interface. `ImageComponent` will include other methods related to images (get, pull, mount)